pkg/commands/patch/transform.go

```diff
@@ -61,6 +72,18 @@ func (self *patchTransformer) transformHeader() []string {
 			"--- a/" + self.opts.FileNameOverride,
 			"+++ b/" + self.opts.FileNameOverride,
 		}
+	} else if self.opts.TurnAddedFilesIntoDiffAgainstEmptyFile {
```

Comment:
What we are doing here seems to be for a very similar purpose as the FileNameOverride for staging, except that the FileNameOverride is for both added and deleted files, but this one we only want to do for added files (I think).

Also, the FileNameOverride drops the `index 1234..5678` line, which is fine for staging and unstaging because there can never be conflicts there. For custom patches it is important to keep this line though (I think).

It might be possible to somehow unify the two mechanisms anyway, but I was too lazy to try.